events/views.py

```diff
@@ -258,3 +263,107 @@ def check_participant_formset(request, event_instance, meeting_participants_form
             event_instance.delete()
             return False
         return True
+
+
+class MeetingVoteView(TemplateView):
+
+    template_name = 'vote/meeting_vote.html'
+
+    def __init__(self, **kwargs) -> None:
+        self.title = None
+        self.vote_form = None
+        self.VoteFormset = None
+        self.chosen_event = None
+        self.meetings_dates = None
+        self.chosen_meeting_dates = None
+        super().__init__(**kwargs)
+
+    def dispatch(self, request, meeting_id, *args, **kwargs):
+        if not request.user.is_authenticated:
+            return redirect(LOGIN_PAGE)
+
+        try:
+            user_id = request.user
+            EventParticipant.objects.get(user_id=user_id, event_id=meeting_id)
+            self.chosen_event = Event.objects.get(id=meeting_id)
+            self.chosen_meeting_dates = OptionalMeetingDates.objects.get_meeting_dates(self.chosen_event)
+            did_user_vote = PossibleParticipant.objects.did_user_vote(self.chosen_meeting_dates, user_id)
+
+            if did_user_vote:
+                return redirect(LOGIN_PAGE)
```

Comment:
Why when a user didn't vote for a meeting he is redirected to the login page? In my opinion, the user should be redirected to the home page.

Reply:
Great catch, 
I have changed the redirection of the users who have voted and who are not an event participant in the meeting to the home page 😄